### ISSUE TITLE
BUG: Negotiating... Timed out

### ISSUE THREAD

**Issue by minepoof** [DiffuseHyperion/InertiaAntiCheat#46]
**Describe the bug**
I'm using a modpack, specifically DB's Minecraft Plus, and I'm trying to login to my server with inertia mod enabled, and I get "Failed to connect to the server Took too long to log in". This happens after it's negotiating for about 20 seconds. Making it unable for me to login to my server.

**To Reproduce**
Steps to reproduce the problem:
1. Download my modpack: 
2. Launch the modpack
3. Login to the server (in my case, mc.cia.cash)
4. Error out

**Expected behavior**
What's supposed to happen is you negotiate then successfully login to the minecraft server and are able to play with the whitelisted mods.

**Screenshots**
![image](https://github.com/user-attachments/assets/b8345cde-7c33-4c39-9ca1-b94bb5d3add0)

**Server information (please complete the following information):**
 - Server software: Fabric
 - Server software loader version: latest
 - Minecraft version: 1.21
 - Mod version: 1.0.3
 - Fabric API version: 0.102.0+1.21

**Mod config**
```# IMPORTANT!
# Please check https://github.com/DiffuseHyperion/InertiaAntiCheat/wiki/Modern-(Version-1.0.0-and-above) on how to configure this mod.
# You will likely need https://iac.diffusehyperion.com/ in the process of configuring this mod.
[mods]
# The method to check client's mod list.
# If you prefer blacklisting/whitelisting certain mods from being used, use "individual".
# If you prefer ensuring client's have a modpack installed, use "group".
# Accepted types: "individual", "group"
method = "individual"
# The hashing algorithm for the mod to use.
# MD5 is the fastest but most insecure.
# SHA256 is the slowest but most secure.
# Accepted algorithms: "MD5", "SHA1", "SHA256"
algorithm = "MD5"
# The kick message to show to clients when they try to connect without InertiaAntiCheat installed.
vanillaKickMessage = "You were kicked for not sending a response to the anti-cheat!\nThis is probably due to missing the anti-cheat client mod.\nInstall it here: https://modrinth.com/mod/inertiaanticheat"
# The kick message to show to clients when they fail the modlist check.
deniedKickMessage = "You were kicked by the anti-cheat!\nThis is due to having illegal mods.\nPlease contact your server administrator for more details."

[mods.individual]
# Immediately kick players who have these mods in their modlist.
# The mod checks (and kicks) for blacklisted mods first.
blacklist = []
# Immediately kick players who do not have these mods in their modlist.
whitelist = ["5f17bf2e1647285058a55710225837b1", "1777f031688fc837c71c04e2c6c67ff8", "2663efaec3087c8940cdc1cffc8c9bf2", "2bd3030631cec891ac17f69413a71426", "166cbbd9af982a82cf493dfde4d347de", "5c0f0f627acd362c874ed046abaf4418", "da74341f2a1a0eea6a61fd76954343ac", "c458e10b8122617a5b9cb1f93758f1cb", "9193a6c55544cb5af013f9d04b5c0d68", "f3e9acdd1c022fc80c27c2f14a70a324", "2c347f63e0757e1b2661cf9e41a22563", "4d4613669ee18e6727195e3eb3d33131", "830219fce0176e2b6a00fa94af2ee7b2", "cd7dbc5efe7f19078ef7609a323e347a", "bbba8c71a368947ee9d66e5f7757a046", "6a037d4e82ee31ac69b80277713ba2f3", "6a037d4e82ee31ac69b80277713ba2f3", "7c3864212203b4d0083f349d047d0ea8", "26b92350ec5dd77d058109ee3fd4a757", "d3cc550b21a8dc12ed5447b28021f086", "3d75857c33a08ddcf35b1ba96fbdbfce", "316cf908b94d508f9f3490bafef39cfc", "12f19b896b7e2eb9556b70a3365c8462", "11cdb65c240251c8cf7c4e8c62e1722b", "4e86f03c440f29ab0cdb1e662b132b1a", "7537674c222dbde8f4402e8f6f056a4b", "3cbfc1a4a3776cc2ba74155150a28394", "8c442abd4f3405f0494aae19f4e51e00", "5ee7f74ec9355f194ddb1ba7f672306f", "7cb3b65417065cc01741ac081c203a11", "4cf57d8c0b73d6a00b22224384953de8", "d7f131cf93da7905271df42828472139", "ebbefb3fc611ed2556f960d419792bc9", "408ad14462ebf8b1e0e01e2a8f788ebd", "bcc5984b38277dc65bb70c0937ab822d", "8c3592449100b0dbee691187ac3c0ba7", "49cfa8b47a361b57f8ff7d934e53cea1", "db92c620cb72a41403b928a0b45955d0", "bc782efced9103fb69ccc1187a83c1e0", "ce2030de5d5bb0ed0d26bcf6bc93f909", "8cade8b884473d574950c8341560118f", "b1bf1e2b0699038478efa6539cfb8ceb", "958a5671cc002bc3fed07b809643be6e", "4c091114556dd911643cef35134e50b4", "34d1411e0789ba6b051c6eabb11eb0bb", "1d39ead1641ef522bc7ae92383f21da8", "139bd33f877532af61edb243fadcf545", "824c1920f22683bef38f2298f3a330c8", "db033d0d00f3af2b45d392197fdf9597", "6e0645e0801e5e07c8a7a2beed2e4af9", "37f06c58105ea1ced930108034c8a9bc", "0d42441da9446c1f10ac4737d73647a1", "93a2726eff1ebdbea3d2ceb8a6e6a498", "b89a2d2bcb6729af37e4a7fba0b706b0", "2d6904baa4c36b0b1f5edf266464345c", "20a781209473e95d66f755bdb7024f2b", "9a40805bf8ad2f6bca0627aa1fe52fa9", "fe211a32bb14da14fdc84cca1d8df56c", "3fbd94e6e359554fffe446a384ffe7c9", "fff83455887d85c6f7ebac7bba24fa8f", "fa0891046cd25a40e6cdf8383ef17147", "774df01e937bcece188ef64144d174f2", "4db24c7f18802858182c251dc1586413", "dbb8a0296453f1a6b6b0214d7fa803d5", "f6afa7bac94101addeb4b2ee34eb828f", "b32b1a21807a7309f713d7bb1eae3023", "60d90ca52ee667ca588d08a998f9ea26", "2188cbc68e9455e3d37b26dbd9d811de"]

[mods.group]
# The hash for the server to check against.
# You can add multiple hashes into the list, and the server will check against all of them.
hash = ["8a692f5b259539b1a9d09aeb76e52141"]
# Allow certain mods to be used by players, in addition to your modpack.
# This allows players to make use of optional client mods if you allow it.
softWhitelist = ["a289e30c5862227a2270a3950a1fe17b"]

[motd]
# Whether to show an icon on client's server list indicating that InertiaAntiCheat is installed.
showInstalled = true
# The names of blacklisted mods to report to clients.
# Clients will be able to see these mod names in their server browser.
# You may choose to put arbituary names, extra names, fake names or no names at all here.
# Setting this to be an empty list will cause the icon to not show up.
blacklist = ["Banned mods: ", "None"]
# The names of whitelisted mods to report to clients.
# Clients will be able to see these mod names in their server browser.
# You may choose to put arbituary names, fake names or no names at all here.
# Setting this to be an empty list will cause the icon to not show up.
whitelist = ["Whitelisted mods: ", "None"]
# The name of the modpack to report to clients.
# Clients will be able to see the modpack name in their server browser.
# Setting this to be an empty list will cause the icon to not show up.
hash = ["Requires modpack: ", "DB's Minecraft Plus"]

[debug]
# Show additional information in server logs.
debug = true
# do not touch pls :)
version = 7```

**Comment by DiffuseHyperion** [comment]
This is probably due to the player taking too long to send every mod (the entire file, not just names) from the modpack to the server. Unfortunately, theres not really a way around this due to minecraft limits but I am working on a new version that will only request the name of each mod, which should eliminate this problem.

**Comment by Tunga-Tech** [comment]
Same

**Comment by konotorii** [comment]
Any updates to this?

**Comment by Tunga-Tech** [comment]
The older versions used to have a grace period in the config but not anymore.

**Comment by DiffuseHyperion** [comment]
Sorry for the 1 month delay 💀, the fix mentioned in the previous comment has been implemented in [version 1.0.4](https://modrinth.com/mod/inertiaanticheat/version/gdSvdQ3F).
